app/controllers/reservations_controller.rb

```diff
@@ -1,33 +1,44 @@
 class ReservationsController < ApplicationController
   before_action :set_screening, only: %i[new create]
+  before_action :set_reservation, only: %i[show update]
+
   def new
     @reservation = Reservation.new
-    @screening = Screening.find(params[:screening_id])
   end
 
   def create
     @reservation = Reservation.new(screening_id: params[:screening_id], status: :booked)
```

Comment:
It would be better to make sure the screening_id actually matches a screening in our database, if you run `set_screening` before this action as well then you could do
```suggestion
    @reservation = @screening.reservations.new(status: :booked)
```
This will make a new instance of reservation with screening_id already set
Generally, assigning raw id's to records is not advised, it's better to find by that id and then assign the object like so:
`Reservation.new(screening: @screening)`
That way you can refer to those assigned objects in other parts of this action if you need to